yarn-project/aztec-node/src/aztec-node/http-node.ts

```diff
@@ -0,0 +1,305 @@
+    proof: tx.proof?.toBuffer().toString('hex'),
+    newContractPublicFunctions: tx.newContractPublicFunctions?.map(f => f.toBuffer().toString('hex')) ?? [],
+    enqueuedPublicFunctions: tx.enqueuedPublicFunctionCalls?.map(f => f.toBuffer().toString('hex')) ?? [],
+  };
+}
+
+/**
+ * Deserialises a transaction from JSON.
+ * @param json - The JSON representation of the transaction.
+ * @returns The deserialised transaction.
+ */
+export function txFromJson(json: any) {
+  const publicInputs = json.data ? KernelCircuitPublicInputs.fromBuffer(Buffer.from(json.data, 'hex')) : undefined;
+  const unverified = json.unverified ? UnverifiedData.fromBuffer(Buffer.from(json.unverified, 'hex')) : undefined;
+  const txRequest = json.txRequest ? SignedTxRequest.fromBuffer(Buffer.from(json.txRequest, 'hex')) : undefined;
+  const proof = json.proof ? Buffer.from(json.proof, 'hex') : undefined;
+  const newContractPublicFunctions = json.newContractPublicFunctions
+    ? json.newContractPublicFunctions.map((x: string) => EncodedContractFunction.fromBuffer(Buffer.from(x, 'hex')))
+    : [];
+  const enqueuedPublicFunctions = json.enqueuedPublicFunctions
+    ? json.enqueuedPublicFunctions.map((x: string) => PublicCallRequest.fromBuffer(Buffer.from(x, 'hex')))
+    : [];
+  if (txRequest) {
+    return Tx.createPublic(txRequest);
+  }
+  if (publicInputs && proof && unverified) {
+    return Tx.createPrivate(
+      publicInputs,
+      Proof.fromBuffer(proof),
+      unverified,
+      newContractPublicFunctions,
+      enqueuedPublicFunctions,
+    );
+  }
+  return Tx.create(publicInputs, proof == undefined ? undefined : Proof.fromBuffer(proof), unverified, txRequest);
+}
+
+/**
+ * A Http client based implementation of Aztec Node.
+ */
+export class HttpNode implements AztecNode {
+  private baseUrl: string;
+  constructor(baseUrl: string) {
+    this.baseUrl = baseUrl.toString().replace(/\/$/, '');
+  }
+  /**
+   * Method to determine if the node is ready to accept transactions.
+   * @returns - Flag indicating the readiness for tx submission.
+   */
+  public async isReady(): Promise<boolean> {
+    const url = new URL(this.baseUrl);
+    const response = await fetch(url.toString());
+    const respJson = await response.json();
+    return respJson.isReady;
+  }
+
+  /**
+   * Method to request blocks. Will attempt to return all requested blocks but will return only those available.
+   * @param from - The start of the range of blocks to return.
+   * @param take - The number of blocks desired.
+   * @returns The blocks requested.
+   */
+  async getBlocks(from: number, take: number): Promise<L2Block[]> {
+    const url = new URL(`${this.baseUrl}/get-blocks`);
+    url.searchParams.append('from', from.toString());
+    if (take !== undefined) {
+      url.searchParams.append('take', take.toString());
+    }
+    const response = await (await fetch(url.toString())).json();
+    const blocks = response.blocks as string[];
```

Comment:
Let's create an issue to add validation to these responses (we can use json-schema, or even go full swagger and autogenerate these wrappers), so we fail early if something goes wrong

Reply:
[693](https://github.com/AztecProtocol/aztec-packages/issues/693)